packages/kbn-apm-synthtrace/src/scenarios/infra_hosts_with_apm_hosts.ts

```diff
@@ -39,28 +40,58 @@ const scenario: Scenario<InfraDocument | ApmFields> = async (runOptions) => {
         );
 
       // APM Simple Trace
+      const instances = times(numInstances).map((index) => {
+        return apm
+          .service({
+            name: `synth-node-${index % 3}`,
+            environment: ENVIRONMENT,
+            agentName: 'node-js',
+          })
+          .instance(`host-${index}`);
+      });
 
-      const instances = [...Array(numServices).keys()].map((index) =>
-        apm
-          .service({ name: `synth-node-${index}`, environment: ENVIRONMENT, agentName: 'nodejs' })
-          .instance('instance')
-      );
       const instanceSpans = (instance: Instance) => {
+        const hasHighDuration = Math.random() > 0.5;
+        const throughput = random(1, 10);
+
+        const traces = range.ratePerMinute(throughput).generator((timestamp) => {
+          const parentDuration = hasHighDuration ? random(1000, 5000) : random(100, 1000);
+          const generateError = random(1, 4) % 3 === 0;
+          const span = instance
+            .transaction({ transactionName })
+            .timestamp(timestamp)
+            .duration(parentDuration);
+
+          return !generateError
+            ? span.success()
+            : span.failure().errors(
+                instance
+                  .error({
+                    message: `No handler for ${transactionName}`,
```

Comment:
I have a similar suggestion here. 

The "No handler" string feels like a magic string. If this is meant to be used in other places, we could create a function to generate these errors based on their type (or any other attribute that makes the most sense).

Reply:
This is basically a test scenario, not really meant to be reused. We could in the future inves time in creating helper functions that will help to have consistent test data that reflect the real world.